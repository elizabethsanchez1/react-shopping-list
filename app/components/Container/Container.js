import PropTypes from 'prop-types';
import React from 'react';
import { View, ScrollView, StyleSheet } from 'react-native';
import theme from '../../styles/theme.style';

const styles = StyleSheet.create( {
  view: {
    flex: 1,
    backgroundColor: theme.PRIMARY_BACKGROUND,
  },
  scrollView: {
    flex: 1,
    backgroundColor: theme.PRIMARY_BACKGROUND,
  },

  altScrollView: {
    backgroundColor: theme.SECONDARY_BACKGROUND,
  },

  altView: {
    backgroundColor: theme.SECONDARY_BACKGROUND,
  },
} );

const Container = ( { scroll, children, alt, containerStyling } ) => {
  if ( alt === true && scroll === true ) {
    return (
      <ScrollView
        contentContainerStyle={ [ styles.altScrollView, containerStyling ] }
        // {...props}
      >
        {children}
      </ScrollView>
    );
  }

  if ( scroll === true ) {
    return (
      <ScrollView
        contentContainerStyle={ [ styles.scrollView, containerStyling ] }
        // {...props}
      >
        {children}
      </ScrollView>
    );
  }


  if ( alt === true ) {
    return (
      <View
        style={ [ styles.altView, containerStyling ] }
        // {...props}
      >
        {children}
      </View>
    );
  }


  return (
    <View
      style={ [ styles.view, containerStyling ] }
      // {...props}
    >
      {children}
    </View>
  );
};

Container.propTypes = {
  alt: PropTypes.bool,
  containerStyling: PropTypes.object,
  scroll: PropTypes.bool,
  children: PropTypes.any,
};

export default Container;
