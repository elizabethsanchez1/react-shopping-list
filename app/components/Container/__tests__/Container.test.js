import 'react-native';
import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Container from "../Container";


describe('Container Component', () => {


  it('Container base visually looks the same', () => {
    const tree = renderer.create(
      <Container>
        <Text>Container</Text>
      </Container>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Container scroll base visually looks the same', () => {
    const tree = renderer.create(
      <Container scroll>
        <Text>Container</Text>
      </Container>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });


});




