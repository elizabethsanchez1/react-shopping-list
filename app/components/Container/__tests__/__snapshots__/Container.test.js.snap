// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component Container base visually looks the same 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#32333D",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text>
    Container
  </Text>
</View>
`;

exports[`Container Component Container scroll base visually looks the same 1`] = `
<RCTScrollView
  contentContainerStyle={
    Array [
      Object {
        "backgroundColor": "#32333D",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View>
    <Text>
      Container
    </Text>
  </View>
</RCTScrollView>
`;
